Runner/JobManager.py

```diff
@@ -0,0 +1,332 @@
+from azure.storage.blob.models import BlobBlock, ContainerPermissions, ContentSettings
+from pathlib import Path
+import azure.batch.models as batchmodels
+import traceback
+import asyncio
+import Utils
+import os
+import datetime
+import functools
+import datetime
+import time
+import CustomTemplateFactory as ctm
+"""
+This module is responsible for creating, submitting and monitoring the pools and jobs
+
+"""
+
+
+_time = str(datetime.datetime.now().day) + "-" + \
+    str(datetime.datetime.now().hour) + "-" + \
+    str(datetime.datetime.now().minute)
+
+
+async def submit_job(batch_service_client, template, parameters):
+    """
+    Submits a Job against the batch service.
+
+    :param batch_client: The batch client to use.
+    :type batch_client: `batchserviceclient.BatchServiceClient`
+    :param str template: the json desciption of the job
+    :param str the parameters of the job
+    """
+    try:
+        loop = asyncio.get_event_loop()
+        job_json = batch_service_client.job.expand_template(
+            template, parameters)
+        jobparameters = batch_service_client.job.jobparameter_from_json(
+            job_json)
+        job = await loop.run_in_executor(None, functools.partial(batch_service_client.job.add, jobparameters))
```

Comment:
Please note: Which ever prefered way you intend to choose for `async` keep it consistent. Since we are upgrading this to `Python 3.7` key update we can use now is the prefered way of handling `coroutine and tasks` https://docs.python.org/3/library/asyncio-task.html 

```
import asyncio
import time

async def say_after(delay, what):
    await asyncio.sleep(delay)
    print(what)

async def main():
    print(f"started at {time.strftime('%X')}")

    await say_after(1, 'hello')
    await say_after(2, 'world')

    print(f"finished at {time.strftime('%X')}")

asyncio.run(main())

```
**Expected output** :

```
started at 12:13:52
hello
world
finished at 12:13:55
```
For gather sample: (To run task concurrently) https://docs.python.org/3/library/asyncio-task.html#running-tasks-concurrently 

```
import asyncio

async def factorial(name, number):
    f = 1
    for i in range(2, number + 1):
        print(f"Task {name}: Compute factorial({i})...")
        await asyncio.sleep(1)
        f *= i
    print(f"Task {name}: factorial({number}) = {f}")

async def main():
    # Schedule three calls *concurrently*:
    await asyncio.gather(
        factorial("A", 2),
        factorial("B", 3),
        factorial("C", 4),
    )

asyncio.run(main())

# Expected output:
#
#     Task A: Compute factorial(2)...
#     Task B: Compute factorial(2)...
#     Task C: Compute factorial(2)...
#     Task A: factorial(2) = 2
#     Task B: Compute factorial(3)...
#     Task C: Compute factorial(3)...
#     Task B: factorial(3) = 6
#     Task C: Compute factorial(4)...
#     Task C: factorial(4) = 24
```

Reply:
I've gone ahead and removed all async functions and replaced them with threading. Found threading to be exactly what I needed to use and easier. Yes under the hood they are similar but the removing all async/await operations made my code easier to understand if someone was to read over the code. 

The whole idea is for every template we want to test we want to create a new Thread where it can be run and wait for it to complete. Sure we can do the same thing with async gather but it more straightforward using a thread pool

Reply:
I'll be adding in some mock tests for the next chunk of work I need to do once this is in the new repo. I'll also need a offline template tester that will compare params json file vs the job json file since this will cause a lot of build issues later down the track. There are some plans to move this branch somewhere else I would like this branch in before the move.